### PR TITLE
feat: implement cachecontrol with no-cache

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/Common/Infrastructure/Controllers/RegistryApiController.cs [hunk 1]
```diff
@@ -36,7 +36,13 @@ public abstract class RegistryApiController<T> : ApiController<T>
             ILogger<T> logger) : base(redis, logger)
         {
             _restClient = restClient;
-            CacheToggle = cacheToggle;
+            _cacheToggle = cacheToggle;
+        }
+
+        protected bool ShouldGetFromCache()
```
Comment (jvandaal) [inline src/Common/Infrastructure/Controllers/RegistryApiController.cs:42]: @ArneD we're not yet convinced about the method name. The default behavior should be that the cache is tried. The default behavior can than be changed through config our a client's request. We were doubting between CanGetFromCache or the current name.

File: src/Common/Infrastructure/Controllers/RegistryApiController.cs [hunk 2]
```diff
@@ -36,7 +36,13 @@ public abstract class RegistryApiController<T> : ApiController<T>
...
+        {
+            return _cacheToggle.FeatureEnabled &&
+                   HttpContext.Request.Headers.CacheControl.ToString().ToLower() != "no-cache";
```
Comment (ArneD) [inline src/Common/Infrastructure/Controllers/RegistryApiController.cs:45]: `HttpContext.Request.Headers.CacheControl.Any(x => x.Contains("no-cache"))` something like this?

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/Common/Infrastructure/Controllers/RegistryApiController.cs (hunk 2 above)
Comment (emalfroy) [reply src/Common/Infrastructure/Controllers/RegistryApiController.cs]: We had to use GetCommaSeparatedValues. This handles following scenario's:
1. Cache-Control: value1, value2 , value3 (redundant spaces)
2. Cache-Control: value1
    Cache-Control: value2 (multiple entries for the same header)
ref. https://github.com/json-api-dotnet/JsonApiDotNetCore/pull/973

File: src/Public.Api/Address/AddressController-Get.cs [hunk 3]
```diff
@@ -1,5 +1,7 @@
 namespace Public.Api.Address
 {
+    using System;
```
Comment (ArneD) [inline src/Public.Api/Address/AddressController-Get.cs:3]: usings?

File: src/Public.Api/Address/AddressController-Get.cs [hunk 4]
```diff
@@ -61,10 +63,10 @@ public partial class AddressController
             var contentFormat = DetermineFormat(actionContextAccessor.ActionContext);
 
             RestRequest BackendRequest() => CreateBackendDetailRequest(objectId);
-
+
```
Comment (ArneD) [inline src/Public.Api/Address/AddressController-Get.cs:66]: whitespace

**Pull request merged**